### DISCUSSION TITLE
304, "cannot use at() with "type_name" - Why cant this string be parsed?

### DISCUSSION THREAD

**Discussion by dominique120** [nlohmann/json#2407] [Q&A] [answered]
I'm trying to parse this json string but I keep getting this error
```
304, "cannot use at() with "type_name"
```
I'm not sure why it never gives me the full error in Visual Studio. I'm using the default methods from the macro.

This is the code:
```
	std::string json_raw = "{\
		\"District\": \"DIstrict 5\",\
		\"FirstName\" : \"John\",\
		\"LastNames\" : \"Doe\",\
		\"PrimaryAddress\" :\"Street #25\",\
		\"PrimaryPhone\" : \"126845136\",\
		\"SecondaryAddress\" : \"\",\
		\"SecondaryPhone\" : \"985148954\"}";

		Client cl;
		nlohmann::json j = json_raw;
		cl.from_json(j, cl);
```

When I use my own parser with the [] operator with a string argument(305). Whats going on there?
```j["ClientId"].get<uint32_t>(),```

I'm using visual studio 2019 and the latest version of this library from vcpkg.

**Comment by nlohmann** [comment, marked as answer]
You did not parse the string, but created a JSON value of type text. Use

```cpp
auto j = nlohmann::json::parse(json_raw);
```

instead.

**Reply by dominique120** [reply]
And how can I map the contents of the parsed json to a class? When I call the "from_json" method(from the macro) on this variable I get the same error.

**Reply by nlohmann** [reply]
It seems you defined `from_json` as a member function. You should not do this. See https://github.com/nlohmann/json#arbitrary-types-conversions.

**Reply by dominique120** [reply]
Still the same error. I have no idea what I'm doing wrong. This is the code https://gist.github.com/dominique120/c50ae284e45436dc58626ab3608154df
I have tried using NON_INTRUSIVE, INTRUSIVE, inside of the class, outside of it, I tried writing my own parse with at(), with the [] operator but nothing works. Everything gives errors or simple does not parse it. What am I doing wrong? Thanks

**Reply by nlohmann** [reply]
You do not need to call the from_json function explicitly. `Client cl = j;` is sufficient.

**Reply by dominique120** [reply]
Now its saying: [json.exception.type_error.302] type must be number, but is string, with your proposed changes. I'm guessing its because I'm sending the integer value as a string, and int this particular case, an empty string. Is there some way of dealing with this while still using the macros?

**Reply by nlohmann** [reply]
No, this must be done in a user-defined function.
